### PR TITLE
Add stub implementations for value set classes

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
This is just add some stub implementation for the value set classes we'll be moving the existing value set functionality to (and fix it along the way).

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: unit/analyses/variable-sensitivity/value_set/pointer_abstract_object.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+TEST_CASE(
+  "A value set pointer abstract object created from type is top",
+  VALUE_SET_TEST_TAGS)
+{
+  value_set_pointer_abstract_objectt abstract_object(
+    pointer_typet(signedbv_typet{32}, 32));
+  REQUIRE(abstract_object.is_top());
+  REQUIRE(!abstract_object.is_bottom());
+}
```
Comment (smowton) [inline unit/analyses/variable-sensitivity/value_set/pointer_abstract_object.cpp]: I suspect CI will give you a hard time for missing end-of-line here and elsewhere

**Review by smowton — APPROVED** [review]
Harmless

File: src/analyses/variable-sensitivity/value_set_abstract_value.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+ Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+/// \file
+/// Value sets for primitives
+
+// NOLINTNEXTLINE(whitespace/line_length)
+#ifndef CPROVER_ANALYSES_VARIABLE_SENSITIVITY_VALUE_SET_ABSTRACT_VALUE_H
```
Comment (danpoe) [inline src/analyses/variable-sensitivity/value_set_abstract_value.h:13]: `NOLINTNEXTLINE` seems unnecessary (here and elsewhere)

File: src/config.inc [hunk 3]
```diff
@@ -5,7 +5,7 @@ BUILD_ENV = AUTO
 ifeq ($(BUILD_ENV),MSVC)
   #CXXFLAGS += /Wall /WX
 else
-  CXXFLAGS += -Wall -pedantic -Werror -Wno-deprecated-declarations -Wswitch-enum
+  CXXFLAGS += -Wall -pedantic -Wno-deprecated-declarations -Wswitch-enum
```
Comment (danpoe) [inline src/config.inc]: I take it that this is because a new compiler version warns about things the older ones did not? Can we use `-Wno-error=warning` to only disable these new warnings being treated as errors? Also should be in a separate commit.

**Review by danpoe — COMMENTED** [review]
Code looks good, though the change disabling `-Werror` should be changed

File: src/config.inc (hunk 3 above)
Comment (hannes-steffenhagen-diffblue) [reply src/config.inc]: This is an accident actually, I need this change only locally because I’m using some newer compiler versions than what we’re running on CI.

File: src/analyses/variable-sensitivity/value_set_abstract_value.h (hunk 2 above)
Comment (hannes-steffenhagen-diffblue) [reply src/analyses/variable-sensitivity/value_set_abstract_value.h:13]: cpplint complains about the line length without these.

**Review by danpoe — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]
